pkg/config/config.go

```diff
@@ -96,9 +99,46 @@ func LoadFromFile(filename string) (*Config, error) {
 		config.LinkKubeconfig = &val
 	}
 
+	if err := config.validate(); err != nil {
+		return nil, err
+	}
```

Comment:
this will break gardenctl if there is a faulty garden config, e.g. one entry does not contain a kubeconfig. This should not be the case. 

In the example below you are not even able to correct the "mistake"
```
g config set-garden foo
Successfully configured garden "foo"
```

```
g config set-garden foo --kubeconfig /foo/bar
Error: failed to complete command options: failed to get target manager: failed to load config: kubeconfig is required for gardens in the gardenctl configuration
```

Similar to the `kubectl config` command, we don't want to be so strict. Maybe we even want to move it to a separate `config validate` command? Let's discuss this